crates/nostr/src/nips/nip11.rs

```diff
@@ -48,6 +50,17 @@ pub async fn get_relay_information_document(
         builder = builder.proxy(Proxy::all(proxy)?);
     }
     let client: Client = builder.build()?;
+    let url = {
+        if url.scheme() == "wss" {
+            let mut http_url = url;
+            http_url
+                .set_scheme("https")
+                .expect("'https' is a valid scheme.");
+            http_url
+        } else {
+            url
+        }
+    };
```

Comment:
Change this (and at line 89) with:

```rust
let url: Url = check_scheme(url)?;
```

And add the new `check_scheme` (or other better name) function:

```rust
fn check_scheme(url: Url) -> Result<Url, Error> {
    let mut url = url;
    match url.scheme() {
        "wss" => url.set_scheme("https").map_err(|_| Error::InvalidScheme)?,
        "ws" => url.set_scheme("http").map_err(|_| Error::InvalidScheme)?,
        _ => (),
    };
    Ok(url)
}
```

I know that `set_scheme` method in that context should never fail but I prefer to avoid using `expect` and `unwrap` in these libraries.

So you also need to add the new error:

```rust
#[derive(Debug, thiserror::Error)]
pub enum Error {
    // ...
    #[error("Invalid scheme")]
    InvalidScheme,
}
```